lib/http.js

```diff
@@ -125,7 +125,41 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
   // update keystore config
   app.post(
     `${routes.keystore}`,
-    _authorizeZcapInvocation(),
+    _authorizeZcapInvocation({
+      getExpectedTarget: async ({req}) => {
+        const {body: {id: configId}} = req;
+        const requestUrl = `${req.protocol}://${req.get('host')}${req.url}`;
+        if(configId !== requestUrl) {
+          throw new BedrockError(
+            'The request URL does not match the configuration ID.',
+            'URLMismatchError', {
+              public: true,
+              configId,
+              requestUrl,
```

Comment:
Does this need an http status code of 400?

Reply:
Technically no, because this ends up as the `cause` in the `onError` handler below.  I'm going to put a 400 here and a comment just to avoid confusion.